### PR TITLE
Greg testing

### PR DESCRIPTION
**Pull request by gkondas**
#22 #21 #23 

## Release Notes

* **New Features**
  * Added evaluation suite for generating prediction indices, task datasets, and model evaluation workflows
  * Introduced composite query evaluation with per-code ROC AUC metrics and probability aggregation
  * Added code sampling utilities for training and evaluation set generation

* **Configuration & Improvements**
  * Updated training defaults with optimized learning rates, improved early stopping, and reduced max steps
  * Enhanced model architecture with configurable dropout regularization
  * Extended dataset support for evaluation metrics including subject IDs and prediction timestamps

* **Documentation**
  * Added evaluation suite instructions

### CODE REVIEW

File: src/every_query/config.yaml [hunk 1]
```diff
@@ -42,25 +61,26 @@ lightning_module:
...
     save_dir: "${trainer.default_root_dir}/loggers"
     offline: False
     id: null # pass correct id to resume experiment!
     anonymous: null # enable anonymous logging
     project: EveryQuery
     log_model: False # upload lightning ckpts
     prefix: "" # a string to put at the beginning of metric keys
-    entity: "payalchandak"
+    entity: "gregkondas9-columbia-university"
+    config:
```
Comment (payalchandak) [inline src/every_query/config.yaml:68]: Nice hack! I think we want to fix this programmatically later

File: src/every_query/tasks.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def build_task_label_matrix(
         [pl.lit(None).alias(query).cast(pl.Boolean) for query in query_codes]
     )
 
-    censor_false = censor_df.filter(not pl.col("censored"))
+    censor_false = censor_df.filter(pl.col("censored") == False)
```
Comment (payalchandak) [inline src/every_query/tasks.py]: good catch, thanks

File: src/every_query/eval.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        )
+
+    return cfg
+
+
+def find_best_checkpoint(run_dir: Path) -> Path:
+    """
+    Use best_model.ckpt if present; otherwise fall back to last.ckpt.
+    """
+    best_model_ckpt = run_dir / "best_model.ckpt"
```
Comment (payalchandak) [inline src/every_query/eval.py]: Reading best from ` run_dir / "best_model.ckpt"` but last from `run_dir / "checkpoints" / "last.ckpt"`

**Review by payalchandak — CHANGES_REQUESTED** [review]

File: src/every_query/eval.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    # 1) Load config from this run
+    cfg = load_run_config(run_dir)
+
+    # Make extra sure we don't overwrite anything by accident
+    cfg.do_overwrite = False
+    cfg.do_resume = False
+    cfg.output_dir = str(run_dir)
+
+    # 2) Make eval reproducible: same seed & matmul precision as training
+    seed = cfg.get("seed", None)
```
Comment (payalchandak) [inline src/every_query/eval.py]: Different runs/models may be trained on different seeds. I think we want to use a stable test seed to make sure we sample the same patients and histories in the test set, rather than using the seed set during training?

File: src/every_query/eval.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    #    - kill Trainer.logger
+    #    - also hard-disable wandb via env as a safety net
+    os.environ.setdefault("WANDB_MODE", "disabled")
+    if "trainer" in cfg and hasattr(cfg.trainer, "logger"):
+        logger.info("Disabling trainer logger for evaluation (no new W&B runs).")
+        cfg.trainer.logger = None
+
+    # 4) Instantiate datamodule, lightning_module, trainer from saved cfg
+    logger.info("Instantiating datamodule...")
+    datamodule = instantiate(cfg.datamodule)
```
Comment (payalchandak) [inline src/every_query/eval.py]: Where are you setting the inference query? Would this not load the training queries into the datamodule?

File: src/every_query/config.yaml [hunk 6]
```diff
@@ -7,7 +7,26 @@ query:
   task_dir: ${oc.env:TASK_DIR}
   sample_times_per_subject: 20 # epochs worth of data
   codes:
-    - MEDS_DEATH
+    - LAB//50976//g/dL//value_[7.2,7.4)
```
Comment (mmcdermott) [inline src/every_query/config.yaml]: Don't add concrete code lists into the github as they are dataset specific.

File: src/every_query/config.yaml [hunk 7]
```diff
@@ -1,13 +1,19 @@
+defaults:
+  - _self_
+  - query_codes: "2_10000_ID"
```
Comment (payalchandak) [inline src/every_query/config.yaml]: What does the notation mean here? Just curious

File: src/every_query/config.yaml [hunk 8]
```diff
@@ -75,8 +82,20 @@ trainer:
...
+      strict: True # whether to crash the training if monitor is not found in the validation metrics
+      check_finite: True # when set True, stops training when the monitor becomes NaN or infinite
+      stopping_threshold: null # stop training immediately once the monitored quantity reaches this
+      divergence_threshold: null # stop training as soon as the monitored quantity becomes worse than this
+      check_on_train_epoch_end: false # set to false so that early stopping is checked after each validation loop
   default_root_dir: ${output_dir}
-  min_epochs: 1 # prevents early stopping
+  min_epochs: 0 # prevents early stopping
   max_epochs: null # We don't control the max epochs, we control the max steps.
   max_steps: 1000000
```
Comment (payalchandak) [inline src/every_query/config.yaml]: probably want to bring this down to 100k?

File: src/every_query/eval.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+import torch
+from hydra.utils import instantiate
+from lightning.pytorch import seed_everything
+from omegaconf import DictConfig, OmegaConf
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def values_as_list(**kwargs) -> list[Any]:
```
Comment (payalchandak) [inline src/every_query/eval.py]: move to utils?

File: src/every_query/eval.py [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        raise NotADirectoryError(f"{model_run_dir} is not a directory")
+
+    # Load training config
+    train_cfg = OmegaConf.load(model_run_dir / "resolved_config.yaml")
+
+    # Nuke the logger so wandb dashboard is clean
+    train_cfg.trainer.logger = ""
+
+    seed = train_cfg.get("seed", 42)
+    if seed is not None:
```
Comment (payalchandak) [inline src/every_query/eval.py:35]: if we are setting a default seed 42, how would seed ever be None

File: src/every_query/eval.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    logger.info("Instantiating trainer...")
+    trainer = instantiate(train_cfg.trainer)
+
+    task_set_dir = Path(cfg.task_set_dir)
+    manifest_df = (
+        pl.read_parquet(cfg.manifest_path)
+        .select(["bucket", "code", "code_slug"])
+        .unique()
+        .sort(["bucket", "code"])
+    )
```
Comment (payalchandak) [inline src/every_query/eval.py]: i assume this is reading a list of codes that we want to evaluate on, is this right? what is a bucket or code_slug?

File: src/every_query/eval.py [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    rows: list[dict[str, Any]] = []
+
+    split = cfg.split
+
+    for row in manifest_df.iter_rows(named=True):
+        bucket = row["bucket"]
+        code = row["code"]
+        code_slug = row["code_slug"]
+
+        task_labels_dir = str(task_set_dir / bucket / code_slug)
```
Comment (payalchandak) [inline src/every_query/eval.py]: is setting code_slug somehow equivalent to specifying a eval code?

File: src/every_query/eval_config.yaml [hunk 13]
```diff
@@ -0,0 +1,23 @@
+defaults:
+  - _self_
+
+# inputs
+model_run_dir: /users/gbk2114/EveryQuery/results/outputs/2025-12-10/17-58-26
+task_parent_dir: /users/gbk2114/eq_stuff/tasks/manual
+task_set_name: cde742daabc4ddc5e79f29102ea14107
```
Comment (payalchandak) [inline src/every_query/eval_config.yaml]: what does hash correspond to?

File: src/every_query/eval_config.yaml [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+eval_code_idx: 0
```
Comment (payalchandak) [inline src/every_query/eval_config.yaml]: what is this indexing?

File: src/every_query/train.py [hunk 15]
```diff
@@ -11,11 +13,32 @@
...
+    Examples:
+        >>> int_prod(2, 3)
+        6
+        >>> int_prod(2, 3.5)
+        7
+        >>> int_prod(2.49, 3)
+        7
+    """
+    return round(x * y)
+
```
Comment (payalchandak) [inline src/every_query/train.py:40]: move to utils?
